blog/2024-01-15-statecharts-whats-in-a-name/index.mdx

```diff
@@ -0,0 +1,112 @@
+---
+title: Statecharts - What’s in a name?
+description: Introducing a new Stately Style feature for quickly applying styles to the names in your state machine including tips and guidlines for naming conventions
+tags: [stately, studio, coding conventions, naming, git, GitHub]
+authors: [kevin]
+date: 2024-01-15
+slug: 2024-01-15-statecharts-whats-in-a-name
+image: /blog/TODO.png
+---
+
+Statecharts and visual diagrams are such a powerful way to organize, and convey information. All of those lovely “boxes and arrows” convey meaningful relationships, indicate sequential order, and direct flows in a way that’s easier to understand quickly. Add to that the ability to embed assets in your diagrams and you’re well on your way towards creating truly expressive, executable software diagrams. But there’s still one thing that statecharts have that should make them easy to understand. Text.
+
+{/* truncate */}
+
+When I say text, I’m largely referring to names. States, events, actors, actions, and guards all have names. These are important since they’re included in the executable code or other [formats that are exported from Studio](https://stately.ai/docs/export-as-code) , functioning as unique ids (or shared) so they’re not just for decoration. Many users often ask us for guidance on how to name things in their state machines and in XState.
+
+Some of these requests relate to semantic meaning in regards to the elements of a statechart. Should a state be named “Awake” or “Waking Up”? Should an event be called “Shuffle the deck” or “Shuffling”? The other questions we get are more cosmetic like, “Should state names start with a capital letter?” or “Should multi-word names be separated by spaces, dashes, or underscores?” The former deserves an answer like “it depends” because it does depened on many things: What is the state really doing? What are the names of the other states or events? Does the state have child states and, if so, what’s their purpose and how are they named? You get the picture. The latter group gets an equally frustrating answer, the same answer I get whenever I travel and ask locals “Do I need to tip in this country?”. And that answer is… “Up to you.”
+
+But don’t worry, I wouldn’t leave you hanging out to dry when it comes to something as hard as naming. Here are some basic guidelines to consider when it comes to naming things in your statecharts:
+
+### Clarity of meaning
+
+Just like with naming anything in code, it’s helpful to pick names that are brief, not generic, yet contain just enough information to be unambiguous. Make names “minimally meaningful”.
+
+State name examples:
+
+- "Idle" instead of "State1"
+- "Door Closed" instead of "Closed"
+- "Payment Processing" instead of "State_Awaiting_Payment"
+
+Event name examples:
+
+- "UserClickedSubmit" instead of "Event1"
+- "TemperatureExceededThreshold" instead of "ThresholdReached"
+- "PaymentReceived" instead of "Event_Payment_Received"
```

Comment:
Nit: lowercase recommended for events, `userClickedSubmit` or `user.submitClicked`

Reply:
Good idea. I changed them.